jax_unirep/utils.py

```diff
@@ -310,14 +311,30 @@ def batch_sequences(seqs: List[str]) -> List[List]:
     return order
 
 
-def get_batch_len(batched_seqs: Iterable[str]) -> Tuple[np.ndarray, List]:
-    """
-    Returns the average length of each batch as well as a full array of batch distribution.
+def right_pad(seqs: List[str], max_len: int):
+    """Pad all seqs in a list to longest length on the right with "-"."""
+    return [seq.ljust(max_len, "-") for seq in seqs]
+
 
-    :param batched_seqs: list of lists of sequences, grouped by length.
+def get_batching_func(
+    xs: np.ndarray, ys: np.ndarray, batch_size: int = 25
+) -> Callable:
+    """
+    Create a function which returns batches of sequences
+    
+    :param xs: array of embedded same-length sequences
+    :param ys: array of one-hot encoded groud truth next-AA labels
     """
-    batch_lens = np.array([len(batch) for batch in batched_seqs])
-    return np.mean(batch_lens), batch_lens
+
+    def batching_func():
+        nonlocal xs, ys
```

Comment:
They were necessary because of a design flaw on my side. I Reassigned the subset of sequences to `xs` and `ys`, and for that you need the `nonlocal` scope. I changed it now so that it doesn't mutate the original variables, so `nonlocal` is not needed anymore.